File: my-alogorithms-study/src/main/scala/com/wenthomas/alogorithms/sort/BubbleSort.scala
package com.wenthomas.alogorithms.sort

/**
 * @author Verno
 * @create 2020-04-11 13:49 
 */
/**
 * 冒泡排序
 */
object BubbleSort {
    def main(args: Array[String]): Unit = {
        val arr = Array(30, 20, 40, 10, 60, 80, 50, 70)
        println(arr.mkString(","))
        val arr1 = sort(arr)
        println(arr.mkString(","))
    }

    def sort(arr: Array[Int]) = {
        for (j <- 0 until arr.length - 1) {
            for (i <- 0 until arr.length - 1 - j) {
                if (arr(i) > arr(i + 1)) {
                    val tmp = arr(i)
                    arr(i) = arr(i + 1)
                    arr(i + 1) = tmp
                }
            }
        }
        arr
    }
}

File: my-alogorithms-study/src/main/java/com/wenthomas/alogorithms/BubbleSort.java
package com.wenthomas.alogorithms;

import java.util.Arrays;

/**
 * 冒泡排序
 * @author Verno
 * @create 2022-11-28 10:02
 */
public class BubbleSort {
    public static int[] bubbleSort(int[] array) {
        if (null != array) {
            int length = array.length;
            for (int i = 0; i < length -1; i++) {
                boolean flag = false;
                for (int j = 0; j < length - 1 - i; j++) {
                    if (array[j] > array[j + 1]) {
                        int temp = array[j];
                        array[j] = array[j + 1];
                        array[j + 1] = temp;
                        flag = true;
                    }
                }
                if (!flag) {
                    break;
                }
            }
            return array;
        }
        return null;
    }

    public static void main(String[] args) {
        int[] array = {3, 1, 5, 7, 4, 4};
        System.out.println("长度= " + array.length);

        System.out.println(Arrays.toString(bubbleSort(array)));

    }
}
